src/utils/value_parsers.rs

```diff
@@ -0,0 +1,9 @@
+use std::error::Error;
+
+/// Parse key:value pair from string, used as a value_parser for Clap arguments
+pub(crate) fn kv_parser(s: &str) -> Result<(String, String), Box<dyn Error + Send + Sync>> {
```

Comment:
How are we parsing the existing headers argument? I assume we already have some code there that we could reuse here.